extensions/objects/templates/FilepathEditor.js

```diff
@@ -172,7 +237,7 @@ oppia.directive('filepathEditor', [
           }
 
           var form = new FormData();
-          form.append('image', $scope.data.metadata.uploadedFile);
+          form.append('image', $scope.generateResampledImageFile());
```

Comment:
What exactly do you mean to validate?

Things that come to mind:

(1) Verify that it looks like valid image data
This will be checked in the backend, not sure how to do this in the front-end.

(2) Verify that the output dimensions are correct
Not sure this is worth it. I think the code necessary to validate will be bigger than the two lines that do the resampling:
          var ctx = canvas.getContext('2d');
          ctx.drawImage(img, 0, 0, dimensions.width, dimensions.height);

The problem is that getting an image's width and height seems to be an asynchronous task (need to load the image into an Image object and listen for the load() event).

In the meantime, I've added a check that generateResampledImageFile() returns a File object, before appending it to the Form.

What do you think?